burnman/planet.py

```diff
@@ -0,0 +1,216 @@
+            self.name = name
+            self.rock = rock
+            self.inner_radius = None
+            self.outer_radius = outer_radius
+            self.thickness = None
+            self.n_slices = n_slices
+            self.n_start = None
+            self.n_end = None
+            self.mass = None
+            assert(n_slices>=6)
+
+    class Layer(LayerBase):
+        """
+        A layer with a constant temperature.
+        """
+        def __init__(self, name, rock, outer_radius, temperature = 300.0, n_slices = 100):
+            Planet.LayerBase.__init__(self, name, rock, outer_radius, n_slices)
+            self.constant_temperature = temperature
+
+        def temperature(self, radius):
+            return self.constant_temperature
+
+    class LayerLinearTemperature(LayerBase):
+        """
+        A layer with a linear temperature profile.
+        """
+        def __init__(self, name, rock, outer_radius, temperature_inner, temperature_outer, n_slices = 100):
+            Planet.LayerBase.__init__(self, name, rock, outer_radius, n_slices)
+            self._temperature_inner = temperature_inner
+            self._temperature_outer = temperature_outer
+
+        def temperature(self, radius):
+            return self._temperature_inner + (self._temperature_outer-self._temperature_inner)*(radius-self.inner_radius)/self.thickness
+
+    def __init__(self, layers, n_max_iterations = 50, verbose = False):
+        """
+        Generate the planet based on the given layers (List of Layer)
+        """
+
+        # sort layers
+        self.layers = sorted(layers, key=lambda x: x.outer_radius)
+
+        # compute thickness, slices, indices, etc.
+        self.radial_slices = np.array([])
+        current_radius = 0.0
+        n = 0
+        for layer in self.layers:
+            layer.inner_radius = current_radius
+            slices = np.linspace(current_radius, layer.outer_radius, layer.n_slices)
+            layer.thickness = layer.outer_radius - layer.inner_radius
+            current_radius = layer.outer_radius
+            slices[0] += 1 # make the boundaries one meter thick
+            layer.n_start = n
+            layer.n_end = n + layer.n_slices
+            n += layer.n_slices
+            self.radial_slices = np.append(self.radial_slices, slices)
+
+        self.radial_slices[0] = 0.0  # overwrite inner-most radius to zero
+        self.radius = current_radius
+
+        # initial pressure guess: linear as a function of radius
+        max_p = 350e9
+        self.pressures = np.array([max_p * (1.0 - r / self.radius) for r in self.radial_slices])
+        self.temperatures = [self.get_layer_by_radius(r).temperature(r) for r in self.radial_slices]
+        self.densities = np.empty_like(self.pressures)
+        self.gravity = np.empty_like(self.pressures)
+        self.bulk_sound_speed = np.empty_like(self.pressures)
+        self.shear_velocity = np.empty_like(self.pressures)
+
```

Comment:
Remove seismic velocities here and in _evaluate_eos. These can be computed after the pressure-density profile is made self-consistent.